subxt/src/constants/constants_client.rs

```diff
@@ -0,0 +1,78 @@
+// Copyright 2019-2022 Parity Technologies (UK) Ltd.
+// This file is dual-licensed as Apache-2.0 or GPL-3.0.
+// see LICENSE for license details.
+
+use super::ConstantAddress;
+use crate::{
+    client::OfflineClientT,
+    error::Error,
+    metadata::{
+        DecodeWithMetadata,
+        MetadataError,
+    },
+    Config,
+};
+use derivative::Derivative;
+
+/// A client for accessing constants.
+#[derive(Derivative)]
+#[derivative(Clone(bound = "Client: Clone"))]
+pub struct ConstantsClient<T, Client> {
+    client: Client,
+    _marker: std::marker::PhantomData<T>,
+}
+
+impl<T, Client> ConstantsClient<T, Client> {
+    /// Create a new [`ConstantsClient`].
+    pub fn new(client: Client) -> Self {
+        Self {
+            client,
+            _marker: std::marker::PhantomData,
+        }
+    }
+}
+
+impl<T: Config, Client: OfflineClientT<T>> ConstantsClient<T, Client> {
+    /// Run the validation logic against some constant address you'd like to access. Returns `Ok(())`
+    /// if the address is valid (or if it's not possible to check since the address has no validation hash).
+    /// Return an error if the address was not valid or something went wrong trying to validate it (ie
+    /// the pallet or constant in question do not exist at all).
+    pub fn validate<Address: ConstantAddress>(
+        &self,
+        address: &Address,
+    ) -> Result<(), Error> {
+        if let Some(actual_hash) = address.validation_hash() {
+            let expected_hash = self
+                .client
+                .metadata()
+                .constant_hash(address.pallet_name(), address.constant_name())?;
+            if actual_hash != expected_hash {
+                return Err(MetadataError::IncompatibleMetadata.into())
```

Comment:
When this fails it would be super useful if the actual pallet constant name is embedded in the error.

This is hard to debug otherwise.

Reply:
Yeahh Andrew raised this too; it's a separate issue somewhere so I didn't want to think about it in this PR!